geoapps/contours/application.py

```diff
@@ -125,28 +159,21 @@ def main(self):
             )
         return self._main
 
-    def update_contours(self):
-        """
-        Assign
-        """
-        if self.data.value is not None:
-            self.export_as.value = (
-                self.data.uid_name_map[self.data.value]
-                + "_"
-                + ContoursDriver.get_contour_string(
-                    self.interval_min,
-                    self.interval_max,
-                    self.interval_spacing,
-                    self.fixed_contours,
-                )
-            )
-
     def update_name(self, _):
         if self.data.value is not None:
             self.export_as.value = self.data.uid_name_map[self.data.value]
         else:
             self.export_as.value = "Contours"
 
+    def is_computational(self, attr):
```

Comment:
Looks like the method is shared with edge_detection app. Consider moving it to the base class PlotSelection2D

Reply:
After looking at it more closely, this call is going to be very expensive as it generates an input file for every parameter in the loop of `get_param_dict`. Why does it have to access the fields of input file?

Reply:
I fixed the efficiency problem by moving INITIALIZER into the class and using that instead of the ui.json file.  I was about to move is_computational into the base class when I remembered that I already put it there to handle the old params class format.  So the child classes that use BaseData params override the definition for use with their params classes.